<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <arg name="name" doc="Robot name; used for namespacing." />

  <!-- bag parameters -->
  <arg name="imu_topic"
      doc="ROS topic broadcasting raw imu data." />
  <arg name="image_topic" 
      doc="ROS topic broadcasting rectified camera #1" />
  <arg name="detection_topic" 
      doc="ROS topic broadcasting detected bounding boxes" />
  <arg name="camera_info_topic" 
      doc="ROS topic broadcasting detected bounding boxes" />
  <arg name="visualize_markers_topic"  default="$(arg name)/orcvio/quadrics" 
      doc="ROS topic broadcasting visualization markers" />
  <arg name="poseout_topic" default="/$(arg name)/orcvio_system/poseout" />

  <!-- optional args -->
  <arg name="machine" default="localhost" />
  <arg name="fixed_frame_id" default="$(arg name)/odom" />
  <arg name="base_frame_id" default="$(arg name)/base" />
  <arg name="publish_tf" default="true" />

  <arg name="evaluation_mode" default="false" 
    doc="Evaluation mode is used to compare the results of different flags" />
  <arg name="visualize_rviz" default="true" />

  <!-- System Manager Nodelet  -->
  <group ns="$(arg name)">
    <node pkg="nodelet" type="nodelet" name="orcvio_system"
      args="standalone orcvio/SystemNodelet "
      >
      <param name="fixed_frame_id" value="$(arg fixed_frame_id)"/>
      <param name="base_frame_id" value="$(arg base_frame_id)"/>
      <param name="publish_tf" value="$(arg publish_tf)" />

      <remap from="~imu" to="$(arg imu_topic)"/>
      <!-- <remap from="~imu" to="/acl_jackal/microstrain/data"/> -->
      <remap from="~cam0_image" to="$(arg image_topic)"/>
      <remap from="~poseout" to="$(arg poseout_topic)" />

      <param name="config_file" value="$(find orcvio)/config/phoenix.yaml"/>
      <param name="result_file" value="$(find orcvio)/cache/temp_rmse.txt"/>


      <param name="load_groundtruth_flag" type="int" value="0"/>
      <!-- <param name="use_left_perturbation_flag" type="int" value="1" /> -->
      <param name="track_image_topic" value="$(arg name)/orcvio/track_image" />

    </node>
  </group>
</launch>
